src/main/java/sawtooth/sdk/client/signing/Secp256k1Context.java

```diff
@@ -0,0 +1,125 @@
+
+/**
+ * A Context for signing and verifying Secp256k1 signatures.
+ */
+public class Secp256k1Context implements Context {
+  /**
+   * Logger.
+   */
+  private static final Logger LOGGER = Logger.getLogger(Context.class.getName());
+
+  /**
+   * The number of bytes in the signature.
+   */
+  private static final int NUM_SIGNATURE_BYTES = 64;
+
+  /**
+   * Half of the number of bytes in the signature.
+   */
+  private static final int HALF_NUM_SIGNATURE_BYTES = 32;
+
+  @Override
+  public final String getAlgorithmName() {
+    return "secp256k1";
+  }
+
+  @Override
+  public final PublicKey getPublicKey(final PrivateKey privateKey) {
+    ECKey privKey = ECKey.fromPrivate(privateKey.getBytes());
+    byte[] publicKey = privKey.getPubKey();
+    return new Secp256k1PublicKey(publicKey);
+  }
+
+  /**
+   * Generate a bitcoin-style compact signature.
+   *
+   * @param privateKey ECKey private key
+   * @param data the raw message bytes
+   * @return the raw signature bytes
+   */
+  private static byte[] generateCompactSig(final ECKey privateKey, final byte[] data) {
+    Sha256Hash hash = Sha256Hash.of(data);
+    ECKey.ECDSASignature sig = privateKey.sign(hash);
+
+    byte[] csig = new byte[NUM_SIGNATURE_BYTES];
+
+    System.arraycopy(Utils.bigIntegerToBytes(sig.r, HALF_NUM_SIGNATURE_BYTES), 0,
+                                             csig, 0, HALF_NUM_SIGNATURE_BYTES);
+    System.arraycopy(Utils.bigIntegerToBytes(sig.s, HALF_NUM_SIGNATURE_BYTES), 0,
+                                             csig, HALF_NUM_SIGNATURE_BYTES, HALF_NUM_SIGNATURE_BYTES);
+
+    return csig;
+  }
+
+  @Override
+  public final String sign(final byte[] data, final PrivateKey privateKey) {
+    ECKey privKey = ECKey.fromPrivate(privateKey.getBytes());
+
+    return Utils.HEX.encode(generateCompactSig(privKey, data));
+  }
+
+  @Override
+  public final boolean verify(final String signature, final byte[] data, final PublicKey publicKey) {
+    X9ECParameters x9ECParameters = SECNamedCurves.getByName(this.getAlgorithmName());
+    ECDomainParameters params = new ECDomainParameters(
+        x9ECParameters.getCurve(),
+        x9ECParameters.getG(),
+        x9ECParameters.getN(),
+        x9ECParameters.getH());
+
+    ECPoint qPoint = x9ECParameters.getCurve().decodePoint(publicKey.getBytes());
+
+    ECDSASigner ecdsaSigner = new ECDSASigner();
+    ECPublicKeyParameters ecPublicKeyParameters = new ECPublicKeyParameters(qPoint, params);
+    ecdsaSigner.init(false, ecPublicKeyParameters);
+
+    byte[] signatureBytes = Utils.HEX.decode(signature);
+
+    byte[] rbytes = Arrays.copyOfRange(signatureBytes, 0, HALF_NUM_SIGNATURE_BYTES);
+    byte[] sbytes = Arrays.copyOfRange(signatureBytes, HALF_NUM_SIGNATURE_BYTES, NUM_SIGNATURE_BYTES);
+
+    BigInteger rSig = new BigInteger(1, rbytes);
+    BigInteger sSig = new BigInteger(1, sbytes);
+
+    byte[] hash = Sha256Hash.of(data).getBytes();
+
+    return ecdsaSigner.verifySignature(hash, rSig, sSig);
```

Comment:
This approach might be diverting around the native implementation that we want to use. 
I think we want to call this method:
https://github.com/bitcoinj/bitcoinj/blob/master/core/src/main/java/org/bitcoinj/core/ECKey.java#L701
Having first instantiated this https://github.com/bitcoinj/bitcoinj/blob/master/core/src/main/java/org/bitcoinj/core/ECKey.java#L516